### ISSUE TITLE
Jest timeout trying to install 'Redis stable' on first suit at CI

### ISSUE THREAD

**Issue by andregosling** [mhassan1/redis-memory-server#26]
Hello. I started using this library today, and was great for me. Whatever, on my github CI that I run my e2e tests, it seems like the installation of redis binary inside the package is only happening when I start my server for the first time, at the first suit, which gives me an error of timeout at jest.

Is there a way to install this together with my yarn install, at the step of install packages?

Thank you a lot.

**Comment by mhassan1** [comment]
It should be installed by `postinstall` during `yarn install` or `npm install`. Questions:
1. Which package manager are you using?
2. Which install command are you running?
3. Can you paste the install output?

**Comment by andregosling** [comment]
I'm using yarn as package manager, and standard command for install: yarn install.

I'm not sure if you want the logs of my yarn install or of the Download at the library, on my first suit, so here it goes both:

`warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/api > apollo-server-express > apollo-server-core > apollo-server-plugin-base@3.7.2: The `apollo-server-plugin-base` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023 and October 22nd 2024, respectively). This package's functionality is now found in the `@apollo/server` package. See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/api > apollo-server-express > apollo-server-core > apollo-server-plugin-base > apollo-server-types@3.8.0: The `apollo-server-types` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023 and October 22nd 2024, respectively). This package's functionality is now found in the `@apollo/server` package. See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/api > @nestjs/cli > fork-ts-checker-webpack-plugin > memfs@3.6.0: this will be v4
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/directus > directus > @directus/api > json2csv@5.0.7: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/directus > directus > @directus/api > nodemailer-mailgun-transport > consolidate@0.15.1: Please upgrade to consolidate v1.0.0+ as it has been modernized with several long-awaited fixes implemented. Maintenance is supported by Forward Email at https://forwardemail.net/ ; follow/watch https://github.com/ladjs/consolidate for updates and release changelog
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/directus > directus > @directus/api > @directus/extensions-sdk > rollup-plugin-styles > @types/cssnano@5.1.0: This is a stub types definition. cssnano provides its own type definitions, so you do not need this installed.
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/api > @graphql-codegen/client-preset > @graphql-codegen/visitor-plugin-common > @graphql-tools/relay-operation-optimizer > @ardatan/relay-compiler > babel-preset-fbjs > @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/api > @graphql-codegen/client-preset > @graphql-codegen/visitor-plugin-common > @graphql-tools/relay-operation-optimizer > @ardatan/relay-compiler > babel-preset-fbjs > @babel/plugin-proposal-object-rest-spread@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-object-rest-spread instead.
warning workspace-aggregator-32eae975-d[58](https://github.com/airtask/backend/actions/runs/6720210815/job/18263289832#step:5:59)f-46b5-b7c3-6a68a0efb1c2 > @airtask/directus > directus > @directus/api > nodemailer-sendgrid > @sendgrid/mail > @sendgrid/client > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/directus > directus > @directus/api > nodemailer-sendgrid > @sendgrid/mail > @sendgrid/client > request > uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/directus > directus > @directus/api > sqlite3 > node-gyp > make-fetch-happen > cacache > @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/directus > directus > @directus/api > nodemailer-sendgrid > @sendgrid/mail > @sendgrid/client > request > har-validator@5.1.5: this library is no longer supported
warning workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/directus > directus > @directus/api > @directus/extensions-sdk > rollup-plugin-styles > cssnano > cssnano-preset-*** > postcss-svgo > svgo > stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
[3/5] Fetching packages...
[4/5] Linking dependencies...
warning "@airtask/eslint-config > eslint-config-standard@17.1.0" has unmet peer dependency "eslint-plugin-n@^15.0.0 || ^16.0.0 ".
warning "workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/api > @nestjs/axios@2.0.0" has unmet peer dependency "axios@^1.3.1".
warning "workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/api > apollo-server-express@3.12.1" has unmet peer dependency "express@^4.17.1".
warning "workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/api > graphql-upload@10.0.0" has incorrect peer dependency "graphql@0.13.1 - 14".
warning "workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/api > ts-loader@9.5.0" has unmet peer dependency "webpack@^5.0.0".
warning "workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/api > @graphql-codegen/cli > @graphql-tools/code-file-loader > @graphql-tools/graphql-tag-pluck > @babel/plugin-syntax-import-assertions@7.22.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "workspace-aggregator-32eae975-d58f-46b5-b7c3-6a68a0efb1c2 > @airtask/directus > directus > @directus/api > @directus/extensions-sdk > rollup-plugin-styles@4.0.0" has incorrect peer dependency "rollup@^2.[63](https://github.com/airtask/backend/actions/runs/6720210815/job/18263289832#step:5:64).0".
warning Workspaces can only be enabled in private projects.
warning Workspaces can only be enabled in private projects.
[5/5] Building fresh packages...
success Saved lockfile.
Done in 279.40s.`

Output of redis-memory-server library at my first suit run:
` FAIL  <SUIT_FILE_NAME> (51.401 s)
  ● Console
    console.log
      Downloading Redis stable: 0 % (0mb / 3.3mb)

      at RedisBinaryDownload.printDownloadProgress (../../node_modules/redis-memory-server/src/util/RedisBinaryDownload.ts:308:15)
  ● <SUIT_NAME>
    thrown: "Exceeded timeout of 30000 ms for a hook.`

**Comment by mhassan1** [comment]
I'm not seeing that the `postinstall` for `redis-memory-server` is running.

Please provide:
1. `yarn -v`
2. `node -v`
3. `redis-memory-server` version

Can you also try running `yarn rebuild redis-memory-server`? I see `Workspaces can only be enabled in private projects.` in the output; are you using Yarn Workspaces, and is `redis-memory-server` a dependency in the root or a workspace?

**Comment by gavinastur** [comment]
We also faced the same issue in a monorepo using npm workspaces. The dependencies are hoisted to the root and the binary is correctly downloaded with the `postinstall` script also in the root. Unfortunately when the sub package runs the tests it looks in it's own `node_modules` for the binary which obviously doesn't exist so tries to download it again causing jest to timeout. Right now our temporary workaround is as follows.
```
 "redisMemoryServer": {
    "downloadDir": "../../node_modules/.cache/redis-memory-server/redis-binaries"
  }
```

**Comment by mhassan1** [comment]
This should be resolved on version 0.9.0.

@andregosling Please open a separate issue for that error.

**Comment by gavinastur** [comment]
@mhassan1 I can confirm that is resolved, thanks for the quick fix! 🚀
